components/gitpod-db/src/typeorm/workspace-db-impl.ts

```diff
@@ -798,6 +813,18 @@ export abstract class AbstractTypeORMWorkspaceDBImpl extends AbstractWorkspaceDB
         return <WorkspaceAndInstance>(res);
     }
 
+    async findPrebuiltWorkspacesByProject(projectId: string): Promise<PrebuiltWorkspace[]> {
+        const repo = await this.getPrebuiltWorkspaceRepo();
+
+        let query = repo.createQueryBuilder('pws');
+        query = query.orderBy('pws.creationTime', 'ASC');
+        query = query.innerJoinAndMapOne('pws.workspace', DBWorkspace, 'ws', 'pws.buildWorkspaceId = ws.id');
+        query = query.andWhere('pws.projectId = :projectId', { projectId })
```

Comment:
Nit: This looks a bit awkward with the mutable variable and frequent re-assignments. Maybe chain like in the other code locations?

```suggestion
        const query = repo.createQueryBuilder('pws')
            .orderBy('pws.creationTime', 'ASC')
            .innerJoinAndMapOne('pws.workspace', DBWorkspace, 'ws', 'pws.buildWorkspaceId = ws.id')
            .andWhere('pws.projectId = :projectId', { projectId });
```